app/objects/c_adversary.py

```diff
@@ -30,15 +28,11 @@ def store(self, ram):
             return self.retrieve(ram['adversaries'], self.unique)
         existing.update('name', self.name)
         existing.update('description', self.description)
-        existing.update('phases', self.phases)
+        existing.update('atomic_ordering', self.atomic_ordering)
         return existing
 
     def has_ability(self, ability):
-        for _, v in self.phases.items():
-            for a in v:
-                if ability.unique == a.unique:
-                    return True
-        return False
+        return ability.ability_id in self.atomic_ordering
```

Comment:
Comparing abilities or unique?  Original code is comparing the `unique` value (uuid+platform+executor).